spock-core/src/main/java/org/spockframework/runtime/extension/builtin/PendingFeatureIfExtension.java

```diff
@@ -18,18 +18,23 @@
   protected void featureConditionResult(boolean result, PendingFeatureIf annotation, FeatureInfo feature) {
     if (result) {
       if (feature.isParameterized()) {
-        feature.addInterceptor(new PendingFeatureIterationInterceptor(annotation.exceptions(), annotation.reason(), PENDING_FEATURE_IF));
+        feature.addInterceptor(new PendingFeatureIterationInterceptor(
+          annotation.exceptions(), annotation.reason(), PENDING_FEATURE_IF,
+          feature.getInterceptors().stream().anyMatch(PendingFeatureIterationInterceptor.class::isInstance)));
       } else {
         feature.getFeatureMethod().addInterceptor(
-          new PendingFeatureInterceptor(annotation.exceptions(), annotation.reason(), PENDING_FEATURE_IF));
+          new PendingFeatureInterceptor(
+            annotation.exceptions(), annotation.reason(), PENDING_FEATURE_IF,
+            feature.getFeatureMethod().getInterceptors().stream().anyMatch(PendingFeatureInterceptor.class::isInstance)));
       }
     }
   }
 
   @Override
   protected void iterationConditionResult(boolean result, PendingFeatureIf annotation, IMethodInvocation invocation) throws Throwable {
     if (result) {
-      new PendingFeatureInterceptor(annotation.exceptions(), annotation.reason(), PENDING_FEATURE_IF).intercept(invocation);
+      new PendingFeatureInterceptor(annotation.exceptions(), annotation.reason(), PENDING_FEATURE_IF, false)
```

Comment:
Why is `secondary` always false here?

Reply:
Hm, well actually in this PR it wouldn't matter, as you cannot apply `@PendingFeatureIf` two times and so there can never be a second one. But in the prepared PR for repeatable `@PendingFeatureIf` I indeed missed as test
```
@PendingFeatureIf({ a == 2 })
@PendingFeatureIf({ a == 2 })
def bar() {
```
where this will bite as when I wrote it, there were no conditions yet that could access the data variables. :-/

The problem is, that I would need to somehow see that there is already one of those in the call chain, as those are not registered.

Hm, I maybe have another idea, what do you think about a special `AssertionError` sub class for pending feature handling?
If successful it would then not throw `AssertionError` but this special error. And this special error will never be translated to aborted, but always forwarded. Then we also do not need the `secondary` flag I think.

Reply:
Yep, I did it with the custom exception, I think this is much cleaner and easier to grasp too and the case here also works fine.